tests/dragonfly/__init__.py

```diff
@@ -37,27 +40,60 @@ class DflyInstance:
     def __init__(self, params: DflyParams, args):
         self.args = args
         self.params = params
-        self.proc = None
+        self.proc: Optional[subprocess.Popen] = None
         self._client: Optional[RedisClient] = None
 
+        self.dynamic_port = False
+        if self.params.existing_port:
+            self._port = self.params.existing_port
+        elif "port" in self.args:
+            self._port = int(self.args["port"])
+        else:
+            # Tell DF to choose a random open port.
+            # We'll find out what port it is using lsof.
+            self.args["port"] = -1
+            self._port = None
+            self.dynamic_port = True
+
```

Comment:
This ignores arguments passed from command line with `--df`, I now can't overwrite anything for tests